src/lib/collision_prevention/CollisionPreventionTest.cpp

```diff
@@ -325,6 +325,74 @@ TEST_F(CollisionPreventionTest, testPurgeOldData)
 	orb_unadvertise(vehicle_attitude_pub);
 }
 
+TEST_F(CollisionPreventionTest, testNoRangeData)
+{
+	// GIVEN: a simple setup condition
+	TestTimingCollisionPrevention cp;
+	hrt_abstime start_time = hrt_absolute_time();
+	mocked_time = start_time;
+	matrix::Vector2f original_setpoint(10, 0);
+	float max_speed = 3;
+	matrix::Vector2f curr_pos(0, 0);
+	matrix::Vector2f curr_vel(2, 0);
+	vehicle_attitude_s attitude;
+	attitude.timestamp = start_time;
+	attitude.q[0] = 1.0f;
+	attitude.q[1] = 0.0f;
+	attitude.q[2] = 0.0f;
+	attitude.q[3] = 0.0f;
+
+	// AND: a parameter handle
+	param_t param = param_handle(px4::params::CP_DIST);
+	float value = 10; // try to keep 10m distance
+	param_set(param, &value);
+	cp.paramsChanged();
+
+	// AND: an obstacle message without any obstacle
+	obstacle_distance_s message;
+	memset(&message, 0xDEAD, sizeof(message));
```

Comment:
Huh? I don't get that.

Reply:
To be honest, I don't know either... I copied the skeleton from another test and did not question that line. I tried running the test without that line and it still passes. Should I remove the memsets from all the test, if the pass without them?

Reply:
I did that in the early tests when I was making sure that the messages were actually getting sent/received correctly. Not sure if it is as critical any more.